shopify/product-reviews-template/api/actions/fetchOrderData.js

```diff
@@ -0,0 +1,85 @@
+import { FetchOrderDataGlobalActionContext } from "gadget-server";
+
+/**
+ * @param { FetchOrderDataGlobalActionContext } context
+ */
+export async function run({ params, logger, api, connections }) {
+  const order = await api.shopifyOrder.maybeFindFirst({
+    filter: {
+      singleUseCode: {
+        equals: params.code,
+      },
+    },
+    select: {
+      id: true,
+      orderNumber: true,
+    },
+  });
+
+  if (order) {
+    let lineItems = await api.shopifyOrderLineItem.findMany({
+      filter: {
+        orderId: {
+          equals: order.id,
+        },
+      },
+      select: {
+        id: true,
+        product: {
+          id: true,
+          title: true,
+          featuredMedia: {
+            file: {
+              url: true,
+              alt: true,
+            },
+          },
+        },
+      },
+    });
+
+    let allLineItems = lineItems;
+
+    while (lineItems.hasNextPage) {
+      lineItems = await lineItems.nextPage();
+      allLineItems = allLineItems.concat(lineItems);
+    }
+
+    const seen = {};
+    const products = [];
+
+    for (const {
+      product: {
+        id,
+        title,
+        featuredMedia: {
+          file: { alt, url },
+        },
+      },
+    } of allLineItems) {
+      if (!seen[id]) {
+        seen[id] = true;
+        products.push({
+          id: id,
+          title: title,
+          image: url,
+          alt,
+        });
+      }
+    }
+
+    await api.internal.shopifyOrder.update(order.id, {
+      singleUseCode: null,
+    });
```

Comment:
Do we want to do this before they have left a review? Seems like something that is set post-review

Reply:
Fixed with new functionality